src/status_im/common/universal_links.cljs

```diff
@@ -68,31 +68,39 @@
   (log/info "universal-links: handling community request  " community-id)
   (navigation/navigate-to cofx :community-requests-to-join {:community-id community-id}))
 
-(rf/defn handle-community
-  [cofx {:keys [community-id]}]
-  (log/info "universal-links: handling community" community-id)
-  (navigation/navigate-to cofx :community {:community-id community-id}))
-
-(rf/defn handle-navigation-to-desktop-community-from-mobile
-  {:events [:handle-navigation-to-desktop-community-from-mobile]}
-  [cofx deserialized-key]
-  (rf/merge
-   cofx
-   {:dispatch [:navigate-to :community-overview deserialized-key]}
-   (navigation/pop-to-root :shell-stack)))
-
-(rf/defn handle-desktop-community
+(rf/defn navigate-to-serialized-community
   [_ {:keys [community-id]}]
   (native-module/deserialize-and-compress-key
    community-id
    (fn [deserialized-key]
-     (rf/dispatch [:chat.ui/fetch-community (str deserialized-key)])
-     (rf/dispatch [:handle-navigation-to-desktop-community-from-mobile (str deserialized-key)]))))
+     (rf/dispatch [:communities/navigate-to-community-overview (str deserialized-key)]))))
+
+(rf/defn navigate-to-community-overview
+  {:events [:communities/navigate-to-community-overview]}
+  [cofx deserialized-key]
+  (if (string/starts-with? deserialized-key "z")
+    (navigate-to-serialized-community cofx deserialized-key)
```

Comment:
we should not make function calls in events, but in effects. cc: @flexsurfer ?

Reply:
I think it's ok. It's just put the result effects from another event here

same as
https://github.com/status-im/status-mobile/blob/349d9da58ffa0eee3708705f40ba0c733e8ae49d/src/status_im/common/universal_links.cljs#L152

Reply:
Actually its a little different, because functions which are being called in `on-handle` are all returning result effects, but here,
We are calling `navigate-to-serialized-community..`, which calls `native-module/deserialize-and-compress-key`, which calls native method `.deserializeAndCompressKey` with uses dispatch as callback (not returned).
So basically we are calling a function, which is returning nil here.